docs/howtos/envelope-object.md

```diff
@@ -4,79 +4,116 @@
 
 This guide provides instructions on propagating a set of resources from the hub cluster to joined member clusters within an envelope object.
 
-## Envelope Object with ConfigMap
+## Why Use Envelope Objects?
 
-Currently, we support using a `ConfigMap` as an envelope object by leveraging a fleet-reserved annotation.
+When propagating resources to member clusters using Fleet, it's important to understand that the hub cluster itself is a fully functional Kubernetes cluster. Without envelope objects, any resource you want to propagate would first be applied directly to the hub cluster, which can cause several critical issues:
 
-To designate a `ConfigMap` as an envelope object, ensure that it contains the following annotation:
+1. **Unintended Side Effects**: Resources like ValidatingWebhookConfigurations, MutatingWebhookConfigurations, or Admission Controllers would become active on the hub cluster, potentially intercepting and affecting hub cluster operations.
+
+2. **Security Risks**: RBAC resources (Roles, ClusterRoles, RoleBindings, ClusterRoleBindings) intended for member clusters could grant unintended permissions on the hub cluster.
+
+3. **Resource Conflicts**: Resources intended for member clusters might conflict with hub cluster resources or disrupt hub services.
+
+4. **Resource Consumption**: Resources intended for member clusters could consume CPU, memory, and storage on the hub cluster.
```

Comment:
actually the real reason is the resourceQuota or limitRange will into effect on the hub cluster which in most cases are not an issue but we want to provide a work around.